test/integration/weighted_cluster_integration_test.cc

```diff
@@ -0,0 +1,187 @@
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, Network::Address::IpVersion::v6) {
+    default_weights_.reserve(upstream_count_);
+    int total_cluster_with_name_count =
+        upstream_count_ - Envoy::RepickClusterFilter::TotalUpstreamClusterWithHeaderCount;
+    // For the simplicity of testing purpose, the default weighted cluster array
+    // starts with weights for clusters with `name` filed and followed by
+    // weights for clusters with `cluster_header` field.
+    std::fill_n(std::back_inserter(default_weights_), total_cluster_with_name_count, 20);
+    std::fill_n(std::back_inserter(default_weights_),
+                Envoy::RepickClusterFilter::TotalUpstreamClusterWithHeaderCount, 30);
+  }
+
+  void createUpstreams() override {
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP2);
+    //  Add fake upstreams
+    for (int i = 0; i < upstream_count_; ++i) {
+      addFakeUpstream(FakeHttpConnection::Type::HTTP2);
+    }
+  }
+
+  void initializeConfig(const std::vector<uint64_t>& weights) {
+    int total_cluster_with_name_count =
+        upstream_count_ - Envoy::RepickClusterFilter::TotalUpstreamClusterWithHeaderCount;
+
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      for (int i = 1; i < upstream_count_; ++i) {
+        auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
+        cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+        cluster->mutable_load_assignment()
+            ->mutable_endpoints(0)
+            ->mutable_lb_endpoints(0)
+            ->mutable_endpoint()
+            ->mutable_address()
+            ->mutable_socket_address()
+            ->set_address(fake_upstreams_[i]->localAddress()->ip()->addressAsString());
+        cluster->set_name(absl::StrFormat(Envoy::RepickClusterFilter::ClusterName, i));
+        ConfigHelper::setHttp2(*cluster);
+      }
+
+      auto* cluster_with_name = bootstrap.mutable_static_resources()->mutable_clusters(0);
+      cluster_with_name->set_name(absl::StrFormat(Envoy::RepickClusterFilter::ClusterName, 0));
+      ConfigHelper::setHttp2(*cluster_with_name);
+
+      config_helper_.addFilter("name: repick-cluster-filter");
+    });
+
+    // Modify route with weighted cluster configuration.
+    config_helper_.addConfigModifier(
+        [this, &total_cluster_with_name_count, &weights](
+            envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+                hcm) {
+          for (int i = 0; i < total_cluster_with_name_count; ++i) {
+            auto* cluster_with_name = hcm.mutable_route_config()
+                                          ->mutable_virtual_hosts(0)
+                                          ->mutable_routes(0)
+                                          ->mutable_route()
+                                          ->mutable_weighted_clusters()
+                                          ->add_clusters();
```

Comment:
Haven't the cluster already been added above?
Why does it need to be added again?